public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -45,11 +47,7 @@ const reduceNamespaces = (filters: RuleFilterState) => {
 const reduceGroups = (filters: RuleFilterState) => {
   return (groupAcc: CombinedRuleGroup[], group: CombinedRuleGroup) => {
     const rules = group.rules.filter((rule) => {
-      if (
-        filters.dataSource &&
-        isGrafanaRulesSource(rule.namespace.rulesSource) &&
-        !rule.queries?.find(({ datasource }) => datasource === filters.dataSource)
-      ) {
+      if (filters.dataSource && isGrafanaRulerRule(rule.rulerRule) && !isQueryingDataSource(rule.rulerRule, filters)) {
```

Comment:
I would change it to `if (filters.dataSource && !(isGrafanaRulerRule(rule.rulerRule) && isQueryingDataSource(rule.rulerRule, filters))) {` 
to filter out rules both wher datasource doesn't match and where we don't have the grafana rule.
Though seems improbable this would ever be an issue

Reply:
So I am basically translating that into:

- If we have a filter.dataSource and the rule is an alerting rule then is should be filtered out.
- If we have a filter.dataSource and the rule is an grafana rule then check if the rule is querying the dataSource.

Is that correct?

Reply:
hmm wait no, your version was correct. This would fitler out cloud alerts as well. My bad :)

Reply:
hmm, I think I need to change it a bit. Right?

Reply:
So we are basically filtering out rules in two different places here witch feels a bit hard to follow.

Reply:
What I wanted to do is defien the case where it's a grafana rule, but only has `promRule` and not `rulerRule`, to filter it out in such case. But that's an edge case and *should never happen* (tm) 

But we **should not** filter out cloud alerting rules here. Datasource filtering for those is done at namespace level. This is because  cloud rules in a given namespace all belong to the same datasource, there is no datasource property on the rule itself.  S oon rule level datasource should not be conisdered  for cloud rules and they should pass.  

So I think the existing code is good enough

Reply:
I will leave it as it is and we probably can clean this code up a little bit when we have everything else in place.

Reply:
Sounds ok?

Reply:
yup :+1:  sorry for muddying the waters here